openhands/controller/agent_controller.py

```diff
@@ -144,8 +144,11 @@ async def report_error(self, message: str, exception: Exception | None = None):
         if exception:
             self.state.last_error += f': {exception}'
         detail = str(exception) if exception is not None else ''
```

Comment:
This line will send to the chat box the rest of exceptions, and that's actually surprising. For example:
<img width="384" alt="Screenshot 2024-10-29 at 10 41 43" src="https://github.com/user-attachments/assets/d0179d7b-9cd3-4efc-91ab-9f2cd15a74fd">

I think the (currently) intended behavior is to not show these, or at least make a generic user-friendly message for them. Or did we really change that intention?

Reply:
The current default behaviour is display errors that are `ErrorObservation`'s (e.g., those called with `report_error`) and a temporary toast for errors sent directly to the WebSocket such as response errors (e.g., those called with `send_error`).

Errors like the one in your screenshot probably shouldn't be displayed but only because the developer should handle it in the code (I doubt OpenHands can do something about internal request errors such as these).

In general, it is in my TODO list to display the errors collapsed by default, and probably truncated if they are super long.

Reply:
Yeah, part of the mess right now can be solved if these are not ErrorObs in the first place. OK, cool.